docs: add comments for frrouting configure file

steveyiyo opened the pull request.

steveyiyo committed:
docs: add comments for frrouting configure file

ching-kuo commented:
@steveyiyo Can you move the PDF file one directory up.
`terrafrom` directory should only include Terraform stuffs
Thanks!

steveyiyo commented:
@ching-kuo Did you mean README.md file?

ching-kuo commented:
Yes

steveyiyo commented:
@ching-kuo done, Could you please check it again?

ching-kuo commented:
@steveyiyo Can you add how to run the Terraform script to setup the environment in the README
Please include

- Downloading/setting up OpenStack credential 
- Link to Terraform installation page 
- How to run Terraform script
- How to destroy the environment

Thanks

steveyiyo committed:
feat: move README.md

steveyiyo commented:
OK, I will do it ASAP.

steveyiyo committed:
docs: add guide about how to run Terraform scripts
- Downloading/setting up OpenStack credential
- Link to Terraform installation page
- How to run Terraform script
- How to destroy the environment

steveyiyo commented:
@ching-kuo I have pushed the guide about how to run terraform scripts.

However, I prefer to create a new MarkDown file to introduce how to run terraform scripts. (It's with FRRouting Guide currently)
Once you merged to the master repo, I will open a new PR to do it.

Thanks!

ching-kuo commented on self-paced-labs/network/frr_bgp/README.md:
````diff
@@ -396,4 +412,51 @@ replace:
+```shell
+# 新增 OpenStack 資訊
+export OS_USERNAME="" # 輸入 OpenStack 的帳號
+export OS_TENANT_NAME="" # 輸入 OpenStack 的帳號
````
OS_TENANT_NAME is the project name in OpenStack

ching-kuo commented on self-paced-labs/network/frr_bgp/README.md:
````diff
@@ -396,4 +412,51 @@ replace:
+接著，我們要設定 OpenStack 的環境資訊，修改並執行以下指令。
+
+```shell
+# 新增 OpenStack 資訊
+export OS_USERNAME="" # 輸入 OpenStack 的帳號
+export OS_TENANT_NAME="" # 輸入 OpenStack 的帳號
+export OS_PASSWORD="" # 輸入 OpenStack 的密碼
+export OS_AUTH_URL="https://openstack.cloudnative.tw:5000" # 輸入 OpenStack 的授權網址，可以在 API Access 中找到
````
We can download the credential file directly from dashboard, which will be easier in most cases.

ching-kuo commented on self-paced-labs/network/frr_bgp/terraform/00_variables.tf:
```diff
@@ -1,5 +1,10 @@
 # Params file for variables
 
+# Configure the OpenStack Provider
+provider "openstack" {
+  region      = "RegionOne"
+}
+
```
Prefer not to hard code this, this should be included in the credential file.

ching-kuo reviewed (changes requested):
Minor comments

steveyiyo committed:
docs: change to use credential file, and introduce how to get it
- Remove the way to access API with account password
- Add how to download credential file from the OpenStack Dashboard

steveyiyo committed:
feat: remove openstack provider in 00_variables.tf

steveyiyo commented:
@ching-kuo Thanks for the feedback. Please check it again.

ching-kuo commented on self-paced-labs/network/frr_bgp/README.md:
````diff
@@ -396,4 +412,48 @@ replace:
+當實驗結束後，我們可以使用以下指令來銷毀環境：
+
+```shell
+terraform apply -destroy
````
```terraform destroy``` seems more common

ching-kuo reviewed (commented):
Minor comment, otherwise LGTM

steveyiyo committed:
docs: replace command "apply -destroy" to "destroy"
add refer documents
